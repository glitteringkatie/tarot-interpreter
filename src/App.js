import React, { useState } from "react";
import logo from "./logo.svg";
import "./App.css";
import MajorArcana from "./lib/major-arcana";

function App() {
  const [arcana, setArcana] = useState("-- major or minor? --");
  const [card, setCard] = useState("-- what value? --");
  // if major arcana, second select has major arcana
  // if minor arcana, second select has 1-King

  const chooseArcana = (e) => {
    setCard("-- what value? --");
    setArcana(e.target.value);
  };

  const majorArcana = [
    "The Fool",
    "The Magician",
    "The High Priestess",
    "The Empress",
    "The Emperor",
    "The Hierophant",
    "The Lovers",
    "The Chariot",
    "Strength",
    "The Hermit",
    "Wheel of Fortune",
    "Justic",
    "The Hanged Man",
    "Death",
    "Temperance",
    "The Devil",
    "The Tower",
    "The Star",
    "The Moon",
    "The Sun",
    "Judgement",
    "The World",
  ];

  const suits = [
    "Ace",
    "Two",
    "Three",
    "Four",
    "Five",
    "Six",
    "Seven",
    "Eight",
    "Nine",
    "Ten",
    "Page",
    "Knight",
    "Queen",
    "King",
  ];

  const cards = arcana === "Major Arcana" ? majorArcana : suits;
  const cardOptions = cards.map((c) => {
    return <option>{c}</option>;
  });

  const sentence = MajorArcana.fool.make(
    "Your future contains {{ adjective }} {{ nouns }}."
  );
  return (
    <div className="App">
      <header className="App-header">
        <div className="card-selector">
          What card did you draw?
          <select value={arcana} onChange={chooseArcana}>
            <option disabled>-- major or minor? --</option>
            <option>Major Arcana</option>
            <option>Swords</option>
            <option>Cups</option>
            <option>Wands</option>
            <option>Pentacles</option>
          </select>
          <select value={card} onChange={(e) => setCard(e.target.value)}>
            {[<option disabled>-- what value? --</option>].concat(cardOptions)}
          </select>
        </div>
        <p>{sentence}</p>
      </header>
    </div>
  );
}

export default App;
