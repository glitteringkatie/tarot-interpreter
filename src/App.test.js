import React from "react";
import { render, fireEvent } from "@testing-library/react";
import App from "./App";

test("fills in the cards for Major Arcana", () => {
  const { getByDisplayValue } = render(<App />);
  const majorMinor = getByDisplayValue("-- major or minor? --");
  expect(majorMinor).toBeInTheDocument();
  fireEvent.change(majorMinor, { target: { value: "Major Arcana" } });
  const card = getByDisplayValue("-- what value? --");
  expect(card).toMatchSnapshot();
});

test("fills in the cards for suits", () => {
  const { getByDisplayValue } = render(<App />);
  const majorMinor = getByDisplayValue("-- major or minor? --");
  expect(majorMinor).toBeInTheDocument();
  fireEvent.change(majorMinor, { target: { value: "Swords" } });
  const card = getByDisplayValue("-- what value? --");
  expect(card).toMatchSnapshot();
});

// test("uses the given card predictions", () => {
//   // mock major-arcana.js, check the magician gets called
//   const { getByDisplayValue } = render(<App />);
//   const majorMinor = getByDisplayValue("-- major or minor? --");
//   fireEvent.change(majorMinor, { target: { value: "Major Arcana" } });
//   const card = getByDisplayValue("-- what value? --");
//   fireEvent.change(card, { target: { value: "The Magician" } });
// });
