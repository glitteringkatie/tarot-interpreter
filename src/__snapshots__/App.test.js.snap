// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fills in the cards for Major Arcana 1`] = `
<select>
  <option
    disabled=""
  >
    -- what value? --
  </option>
  <option>
    The Fool
  </option>
  <option>
    The Magician
  </option>
  <option>
    The High Priestess
  </option>
  <option>
    The Empress
  </option>
  <option>
    The Emperor
  </option>
  <option>
    The Hierophant
  </option>
  <option>
    The Lovers
  </option>
  <option>
    The Chariot
  </option>
  <option>
    Strength
  </option>
  <option>
    The Hermit
  </option>
  <option>
    Wheel of Fortune
  </option>
  <option>
    Justic
  </option>
  <option>
    The Hanged Man
  </option>
  <option>
    Death
  </option>
  <option>
    Temperance
  </option>
  <option>
    The Devil
  </option>
  <option>
    The Tower
  </option>
  <option>
    The Star
  </option>
  <option>
    The Moon
  </option>
  <option>
    The Sun
  </option>
  <option>
    Judgement
  </option>
  <option>
    The World
  </option>
</select>
`;

exports[`fills in the cards for suits 1`] = `
<select>
  <option
    disabled=""
  >
    -- what value? --
  </option>
  <option>
    Ace
  </option>
  <option>
    Two
  </option>
  <option>
    Three
  </option>
  <option>
    Four
  </option>
  <option>
    Five
  </option>
  <option>
    Six
  </option>
  <option>
    Seven
  </option>
  <option>
    Eight
  </option>
  <option>
    Nine
  </option>
  <option>
    Ten
  </option>
  <option>
    Page
  </option>
  <option>
    Knight
  </option>
  <option>
    Queen
  </option>
  <option>
    King
  </option>
</select>
`;
